gwdetchar/scattering/core.py

```diff
@@ -107,3 +109,47 @@ def get_fringe_frequency(series, multiplier=2.0):
                         velocity.sample_rate.value)
     fringef.override_unit('Hz')
     return fringef
+
+
+def get_blrms(series, flow=4.0, fhigh=10.0, stride=1, fftlength=4,
+              overlap=2, **kwargs):
+    """ Generate blrms for a TimeSeries
+
+    Parameters
+    ----------
+    series  : `~gwpy.timeseries.TimeSeries` of the channel
+            for which blrms are to be computed.
+    low : `float`
+        lower limit of the bandpass frequency.
+    high : `float`
+         upper limit of the bandpass frequency.
+    stride: `float`
+           time duration of the stride for rms
+
+    Returns
+    -------
+    blrms : A whitened band passed time series.
+     """
+    wseries = series.whiten(fftlength=fftlength, overlap=overlap, **kwargs)
+    bpseries = wseries.bandpass(flow, fhigh)
+    return bpseries.rms(stride)
+
+
+def get_segments(series, threshold, name=None):
+    """ Generates segments from a series that are above
+        some threshold
+
+    Parameters
+    ----------
+    series  : `~gwpy.timeseries.TimeSeries` of the channel
+            for which the segments are to be computed.
+    name : name of the flag for the segments.
+
+    Returns
+    --------
+    DataQualityFlag
+    """
```

Comment:
@siddharth101, can you clean up this docstring?

```python
    """Generate data-quality segments by thresholding a `TimeSeries`

    Parameters
    ----------
    series  : `~gwpy.timeseries.TimeSeries`
        the input `TimeSeries` data

    threshold : `float`
        the threshold value for active data-quality segments

    name : `str`, optional
        name of the data-quality flag, defaults to `series.name`

    Returns
    --------
    threshflag : `~gwpy.segments.DataQualityFlag`
        the populated data-quality flag
    """
```

Reply:
Done.